### PR TITLE
Periodic cleanup of gossip raw message store

### PR DESCRIPTION
**Pull request by sandipbhoir**
## Description
This PR addressess unbounded growth of below two storages:

- **m_rumorHashRawMsgBimap:** maps rumor hash to the full rumor message content
- **m_rumorIdHashBimap:** maps sequential rumorId integers to rumor message hashes

Below two need not to considered:
- **m_tmpRumorHashSet:** temporary rumor hash store to ensure duplicate rumors are not
distributed - **No more used in our codebase.**
- **m_hashesSubscriberMap:** if a node is asked for a message he does not have, the asking
node gets subscribed to this message (entries inside this map get deleted once subscribed
messages are delivered) - **which is good enough to control the growth of this container** 

## Review Suggestion
Please check the file changed.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libNetwork/RumorManager.cpp [hunk 1]
```diff
@@ -516,4 +539,31 @@ void RumorManager::PrintStatistics() {
                               << " ], " << state);
     }
   }
+}
+
+void RumorManager::CleanUp() {
+  int count = 0;
+  auto now = std::chrono::high_resolution_clock::now();
+  while (not m_rumorRawMsgTimestamp.empty()) {
```
Comment (ansnunez) [inline src/libNetwork/RumorManager.cpp]: Please use `!` instead of `not`

File: src/libNetwork/RumorManager.cpp [hunk 2]
```diff
@@ -152,6 +159,11 @@ bool RumorManager::Initialize(const std::vector<Peer>& peers,
     m_rumorHolder.reset(new RRS::RumorHolder(m_peerIdSet, 0));
   }
 
+  RAW_MESSAGE_EXPIRY_IN_MS = (KEEP_RAWMSG_FROM_LAST_N_ROUNDS < MAX_TOTAL_ROUNDS)
```
Comment (ansnunez) [inline src/libNetwork/RumorManager.cpp]: Maybe add a comment to explain the values used here?

File: src/libNetwork/RumorManager.cpp [hunk 3]
```diff
@@ -516,4 +539,31 @@ void RumorManager::PrintStatistics() {
...
+
+void RumorManager::CleanUp() {
+  int count = 0;
+  auto now = std::chrono::high_resolution_clock::now();
+  while (not m_rumorRawMsgTimestamp.empty()) {
+    auto elapsed_milliseconds =
+        std::chrono::duration_cast<std::chrono::milliseconds>(
+            now - m_rumorRawMsgTimestamp.front().second)
+            .count();
+    LOG_GENERAL(DEBUG, "elapsed_milliseconds:" << elapsed_milliseconds
```
Comment (ansnunez) [inline src/libNetwork/RumorManager.cpp]: Will this log too many messages?  If yes, should we replace with
```
if (DEBUG_LEVEL >= 5) {
        LOG_GENERAL(INFO, ...);
}
```

File: src/libNetwork/RumorManager.cpp [hunk 4]
```diff
@@ -196,18 +208,25 @@ bool RumorManager::AddRumor(const RumorManager::RawBytes& message) {
       m_rumorIdHashBimap.insert(
           RumorIdRumorBimap::value_type(++m_rumorIdGenerator, hash));
 
-      m_rumorHashRawMsgBimap.insert(
-          RumorHashRumorBiMap::value_type(hash, message));
+      std::pair<RumorHashRumorBiMap::iterator, bool> result =
```
Comment (ShengguangXiao) [inline src/libNetwork/RumorManager.cpp]: Can use auto result = .....

File: src/libNetwork/RumorManager.h [hunk 5]
```diff
@@ -58,13 +58,18 @@ class RumorManager {
   RumorHashesPeersMap m_hashesSubscriberMap;
   Peer m_selfPeer;
   std::vector<RawBytes> m_bufferRawMsg;
+  std::deque<std::pair<RumorHashRumorBiMap::iterator,
```
Comment (ShengguangXiao) [inline src/libNetwork/RumorManager.h]: Can define a alias type for this?  It is really quite long

File: src/libNetwork/RumorManager.h [hunk 6]
```diff
@@ -58,13 +58,18 @@ class RumorManager {
...
+                       std::chrono::high_resolution_clock::time_point>>
+      m_rumorRawMsgTimestamp;
 
   int64_t m_rumorIdGenerator;
   std::mutex m_mutex;
   std::mutex m_continueRoundMutex;
   std::atomic<bool> m_continueRound;
   std::condition_variable m_condStopRound;
 
+  int32_t RAW_MESSAGE_EXPIRY_IN_MS;
```
Comment (ShengguangXiao) [inline src/libNetwork/RumorManager.h]: If it is not a constant value, it is better use m_rawMessageExpiryInMs

File: src/libNetwork/RumorManager.cpp (hunk 1 above)
Comment (sandipbhoir) [reply src/libNetwork/RumorManager.cpp]: Fixed.

**Pull request merged**